src/auth-service/models/Inquiry.js

```diff
@@ -37,42 +45,75 @@ const InquirySchema = new mongoose.Schema(
 );
 
 InquirySchema.statics = {
-  register(args) {
+  async register(args) {
     try {
+      let modifiedArgs = Object.assign({}, args);
+      const eitherFirstOrLastName = args.firstName
+        ? args.firstName
+        : args.lastName;
+      if (isEmpty(args.fullName) && !isEmpty(eitherFirstOrLastName)) {
```

Comment:
resolve runtimes and refactor the entity logic for the `create inquiry` use case.